src/controlplane/client/connector/connector.go

```diff
@@ -30,27 +30,47 @@ type ConnParams struct {
 	Client client.HTTPDoer
 }
 
-type defaultConnector struct {
-	// TODO: Use a non-sdk logger
-	logger        log.Logger
-	authenticator authenticator.Authenticator
-	endpoints     []config.Endpoint
+type Config struct {
+	Authenticator authenticator.Authenticator
+	Endpoints     []config.Endpoint
 }
 
-// DefaultConnector returns a defaultConnector that probes all
-// endpoints in the list and return the first responding status OK.
-func DefaultConnector(endpoints []config.Endpoint, authenticator authenticator.Authenticator, logger log.Logger) Connector {
-	return &defaultConnector{
-		logger:        logger,
-		authenticator: authenticator,
-		endpoints:     endpoints,
+type OptionFunc func(dc *DefaultConnector) error
+
+// WithLogger returns an OptionFunc to change the logger from the default noop logger.
+func WithLogger(logger log.Logger) OptionFunc {
+	return func(dc *DefaultConnector) error {
+		dc.logger = logger
+		return nil
+	}
+}
+
+// DefaultConnector implements Connector interface for the Control Plane components.
+type DefaultConnector struct {
+	Config
+	logger log.Logger
+}
+
+// New returns a DefaultConnector.
+func New(config Config, opts ...OptionFunc) (*DefaultConnector, error) {
```

Comment:
I am not a super fan of this, since the logic is basically the same as before, but more code is needed.

I imagine we introduce it since after moving to a new package structure

Reply:
we ask to provide config, but then in the .New we simply save config and apply logger..

Reply:
yes I introduced in order to unify a bit the way of objects are created, didn't make much sense to have it different between authenticator and connector IMO. 
I personally like this approach , is easy to extend and to refactor when changes are introduced.